rasa/cli/data.py

```diff
@@ -410,27 +418,35 @@ def _migrate_model_config(args: argparse.Namespace) -> None:
 
     rule_output_file = _get_rules_path(Path(args.out))
 
-    # TODO:
-    # 2. Add telemetry
-    # 3. Auto backup file?
     (
         model_configuration,
         domain,
         new_rules,
     ) = rasa.core.config.migrate_mapping_policy_to_rules(model_configuration, domain)
 
+    if new_rules:
+        _backup(domain_file)
+        domain.persist_clean(domain_file)
```

Comment:
why not do this at the end with the config?

Reply:
this saved me one additional check 😀Added the check and moved this to the end of the function so all dumping things are at the end.